src/internal/schema.ts

```diff
@@ -70,6 +70,22 @@ export function _buildSchema<T, U extends AnyParamConstructor<T>>(
 
   sch.loadClass(cl);
 
+  // Override getters & setters.
+  const accessors = Object.getOwnPropertyDescriptors(cl.prototype);
+  for (const key in accessors) {
```

Comment:
getters & setters are handled by `sch.loadClass` and so shouldnt be modified outside of it
(created an issue https://github.com/Automattic/mongoose/issues/8892)